### PR TITLE
Changed all MLContext creation to include a fixed seed

### PR DESCRIPTION
**Pull request by harishsk**
Many tests are creating MLContext without a seed. Since we are seeing a number of random failures, this PR attempts to standardize all MLContext creation to include a fixed seed. 

Note to reviewers: If you know if cases where a fixed seed is not desired, please flag those cases.

### CODE REVIEW

File: test/Microsoft.ML.TimeSeries.Tests/TimeSeriesSimpleApiTests.cs [hunk 1]
```diff
@@ -169,7 +169,7 @@ public void SpikeDetection()
         [Fact]
         public void SsaSpikeDetection()
         {
-            var env = new MLContext();
+            var env = new MLContext(1);
```
Comment (justinormont) [inline test/Microsoft.ML.TimeSeries.Tests/TimeSeriesSimpleApiTests.cs]: As discussion topic, does any know of a reason to not have a fixed seed for unit tests?

**Comment by najeeb-kazmi** [comment]
>                     new SdcaMaximumEntropyMulticlassTrainer.Options { MaximumNumberOfIterations = 100, Shuffle = true, NumberOfThreads = 1, }));

Shuffle = false for SDCA will give more/full determinism as Justin suggests

---
Refers to: test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs:34 in 4492586. [](commit_id = 44925863ed727b66f344ac7c4cd90bd07528d0ee, deletion_comment = False)

**Comment by najeeb-kazmi** [comment]
>                     new SdcaMaximumEntropyMulticlassTrainer.Options { MaximumNumberOfIterations = 100, Shuffle = true, NumberOfThreads = 1 }))

Same here

---
Refers to: test/Microsoft.ML.Tests/Scenarios/Api/Estimators/Extensibility.cs:45 in 4492586. [](commit_id = 44925863ed727b66f344ac7c4cd90bd07528d0ee, deletion_comment = False)

**Comment by harishsk** [comment]
> Same here, and everywhere else SDCA shows up in tests.
> 

What about cases where Shuffle has been set to true explicitly? E.g.

https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/DecomposableTrainAndPredict.cs#L36

https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/Extensibility.cs#L45

https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs#L34

https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs#L83

**Comment by najeeb-kazmi** [comment]
> > Same here, and everywhere else SDCA shows up in tests.
> 
> What about cases where Shuffle has been set to true explicitly? E.g.
> 
> https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/DecomposableTrainAndPredict.cs#L36
> 
> https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/Extensibility.cs#L45
> 
> https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs#L34
> 
> https://github.com/harishsk/machinelearning/blob/44925863ed727b66f344ac7c4cd90bd07528d0ee/test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs#L83

The default is true, so I don't why it has been set explicitly. I don't see anything in the tests that would require shuffling either.

**Comment by harishsk** [comment]
Would you recommend changing those to false as well in that case?

**Comment by justinormont** [comment]
> Would you recommend changing those to false as well in that case?

I'm unsure if disabling shuffling with cause the unit tests to be more stable. The shuffling and TrainTest/CV splits (hopefully) listen to seed, and would then be deterministic. Can verify by inspecting the one of the TrainTest splits, or repeatedly dumping the iDataView to a file (expecting with a fixed seed they stable perfectly the same, and changing the seed gets you a different set and ordering of rows).

**Comment by harishsk** [comment]
How about we do this in two passes then? Can we merge these changes in first and then do the shuffle=false in a second pass?

**Comment by frank-dong-ms** [comment]
Do you plan to do some test whether this will improve stability?

**Comment by najeeb-kazmi** [comment]
> How about we do this in two passes then? Can we merge these changes in first and then do the shuffle=false in a second pass?

Sounds good to me. Should we also add "seed=1" to the predictor tests in another PR or do you want to take care of this here?

**Comment by harishsk** [comment]
> 
> 
> Do you plan to do some test whether this will improve stability?

By all consensus, seeded unit tests are the right thing to do. The tests passed here on first attempt. So I don't think this has made it any worse. I plan to run full tests against these also.

**Pull request merged**